### PR TITLE
adding .codeclimate.yml configuration

### PR DESCRIPTION
**Pull request by andytinkham**
Signed-off-by: Andy Tinkham <andy.tinkham@cyberark.com>

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -1,15 +1,167 @@
...
+        ruby:
+        swift:
+        typescript:
 
+  fixme: # Flags any FIXME, TODO, BUG, XXX, HACK comments so they can be fixed
+    enabled: true
+    config:
+      strings:
+      - FIXME
+      - TODO
```
Comment (sgnn7) [inline .codeclimate.yml:31]: @izgeri Did we reach a consensus on these?

File: .codeclimate.yml [hunk 2]
```diff
@@ -1,15 +1,167 @@
-engines:
-  golint:
+# This is our default .CodeClimate.yml, broken out by language. Uncomment the
+# sections at the bottom that apply to your project. ACTION comments indicate
+# places where config might need to be tweaked.
+
+version: "2"
+
+plugins:
```
Comment (sgnn7) [inline .codeclimate.yml:7]: This probably needs those changes that @BradleyBoutcher applied to the broker about return statements and method lines

File: .codeclimate.yml [hunk 3]
```diff
@@ -1,15 +1,167 @@
...
+      languages:
+        go:
+        java:
+        javascript:
+        php:
+        python:
+          python_version: 3  # ACTION Comment this out if using Python 2
+        ruby:
+        swift:
+        typescript:
```
Comment (sgnn7) [inline .codeclimate.yml:24]: This repo will never contain some of these so they should be removed:
- java
- php
- swift

File: .codeclimate.yml [hunk 4]
```diff
@@ -1,15 +1,167 @@
...
+      - "**/"         # exclude all
+      - "!./cmd/"     # ynexclude just the ones we want to lint
+      - "!./pkg/"
+      - "!./templates"
+  golint:
+    enabled: true
+  govet:
+    enabled: true
 
+  # Ruby
```
Comment (sgnn7) [inline .codeclimate.yml:57]: All of the ruby checkers should get removed

File: .codeclimate.yml [hunk 5]
```diff
@@ -1,15 +1,167 @@
...
+
+  # CSS
+  csslint:
     enabled: true
 
+  # Groovy
+  codenarc:
+    enabled: true
+
+  # Java
```
Comment (sgnn7) [inline .codeclimate.yml:90]: Needs removal

File: .codeclimate.yml [hunk 6]
```diff
@@ -1,15 +1,167 @@
...
+  pmd:
+    enabled: true
+  sonar-java:
+    enabled: true
+    config:
+      sonar.java.source: "7" # ACTION set this to the major version of Java used
+  # ACTION uncomment checkstyle below if Java code exists in repo
+
+  # Node.js
```
Comment (sgnn7) [inline .codeclimate.yml:99]: Needs removal

File: .codeclimate.yml [hunk 7]
```diff
@@ -1,15 +1,167 @@
...
+  nodesecurity:
+    enabled: true
+  # ACTION uncomment eslint below if JavaScript already exists and .eslintrc
+  # file exists in repo
+
+  # PHP
```
Comment (sgnn7) [inline .codeclimate.yml:105]: Needs removal

File: .codeclimate.yml [hunk 8]
```diff
@@ -1,15 +1,167 @@
...
+      standards: "PSR1,PSR2,WordPress,WordPress-Core,WordPress-Extra"
+  phpmd:
+    enabled: true
+    config:
+      file_extensions: "php,inc,lib"
+      rulesets: "cleancode,codesize,controversial,naming,unusedcode"
+  sonar-php:
+    enabled: true
+
+  # Python
```
Comment (sgnn7) [inline .codeclimate.yml:124]: Needs removal

File: .codeclimate.yml [hunk 9]
```diff
@@ -1,15 +1,167 @@
...
+    enabled: true
+  radon:
+    enabled: true
+    # config:
+    #   python_version: 2 # ACTION Uncomment these 2 lines if using Python 2
+  sonar-python:
+    enabled: true
+
+# ---------------
+# Configuration Required Language specific - these will error and abort the
```
Comment (sgnn7) [inline .codeclimate.yml:137]: This whole section can get removed

File: .codeclimate.yml [hunk 10]
```diff
@@ -1,15 +1,167 @@
...
+
+# ---------------
+# Other languages - will work with or without language files present. Again,
+# runtime is minimal, so OK to leave uncommented.
+
+  # CoffeeScript
+  coffeelint:
+    enabled: true
+
+  # CSS
```
Comment (sgnn7) [inline .codeclimate.yml:82]: Needs removal

File: .codeclimate.yml [hunk 11]
```diff
@@ -1,15 +1,167 @@
...
+
+  # Shell scripts
   shellcheck:
+   enabled: true
+
+# ---------------
+# Other languages - will work with or without language files present. Again,
+# runtime is minimal, so OK to leave uncommented.
+
+  # CoffeeScript
```
Comment (sgnn7) [inline .codeclimate.yml:78]: Needs removal

File: .codeclimate.yml [hunk 12]
```diff
@@ -1,15 +1,167 @@
...
+        javascript:
+        php:
+        python:
+          python_version: 3  # ACTION Comment this out if using Python 2
+        ruby:
+        swift:
+        typescript:
 
+  fixme: # Flags any FIXME, TODO, BUG, XXX, HACK comments so they can be fixed
+    enabled: true
```
Comment (andytinkham) [inline .codeclimate.yml:27]:
```suggestion
    enabled: true
    issue_override:
       severity: info  # Don't fail PRs for FIXME tags, but still flag them
```

File: .codeclimate.yml (hunk 3 above)
Comment (andytinkham) [reply .codeclimate.yml:24]: We're not customizing the engines per repo - the goal is one master file everywhere. All of these engines add minimal time to the scan and cause no issues to run.

File: .codeclimate.yml (hunk 1 above)
Comment (andytinkham) [reply .codeclimate.yml:33]: I updated the config to make them informational.

File: .codeclimate.yml (hunk 4 above)
Comment (andytinkham) [reply .codeclimate.yml:59]: See above

**Comment by andytinkham** [comment]
Need to have broader conversation about return counts and lines of code as we can't make per-language settings. Will reopen after that occurs.

**Pull request closed without merging**